File: api/src/main/java/fr/huiitre/tools/modules/core/auth/infrastructure/password/BCryptPasswordHasher.java
package fr.huiitre.tools.modules.core.auth.infrastructure.password;

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

import fr.huiitre.tools.modules.core.auth.application.ports.PasswordHasher;

public class BCryptPasswordHasher implements PasswordHasher {

    private final BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();

    @Override
    public String hash(String rawPassword) {
        return encoder.encode(rawPassword);
    }

    @Override
    public boolean matches(String rawPassword, String hashedPassword) {
        return encoder.matches(rawPassword, hashedPassword);
    }
}

File: api/Modules/Core/Auth/Infrastructure/Password/BCryptPasswordHasher.cs
using Tools.Api.Modules.Core.Auth.Application.Ports;
using Tools.Api.Modules.Core.Auth.Application.Ports.Password;

namespace Tools.Api.Modules.Core.Auth.Infrastructure.Password;

// Adaptateur technique BCrypt du port IPasswordHasher.
// Format compatible avec les hash produits par l'API Java (Spring BCryptPasswordEncoder).
public sealed class BCryptPasswordHasher : IPasswordHasher
{
    public string Hash(string password) => BCrypt.Net.BCrypt.HashPassword(password);

    public bool Matches(string password, string hash) => BCrypt.Net.BCrypt.Verify(password, hash);
}
